sklearn/cross_decomposition/_pls.py

```diff
@@ -232,6 +232,11 @@ def fit(self, X, Y):
         # paper.
         Y_eps = np.finfo(Yk.dtype).eps
         for k in range(n_components):
+            if np.all(np.dot(Yk.T, Yk) < Y_eps):
```

Comment:
Shouldn't we catch the StopIteration (in the nipals algorithm above) instead?

This is potentially a pretty expensive check to run (matrix  multiplication), and I'm not sure at first sight that it's more accurate?

Reply:
Yea that makes sense. I moved the logic to catching the exception and breaking out of the loop then.